idl/github.com/uber/cadence/shared.thrift

```diff
@@ -713,6 +717,7 @@ struct PollForActivityTaskRequest {
   10: optional string domain
   20: optional TaskList taskList
   30: optional string identity
+  40: optional TaskListMetadata taskListMetadata
```

Comment:
I'm confused how does task list relate to a workflow execution? Multiple workflow types can share a single task list. The same for activity task list. Last time we talked the task list throttling metadata belonged to Domain configuration.

Reply:
@mfateev this is about rate limiting the tasks dispatched from tasklist.  If user wants to rate limit individual activity type they can create separate tasklist for them.

Reply:
Code reviewer made it look this field belonged to WorkflowExecution structure. I still don't believe that this belongs to Poll request as it is not clear what semantic would be of multiple workers passing different values. Also we don't want to redeploy workers to change the rate. IMHO it should be part of Domain metadata.

Reply:
Not sure what you mean by WorkflowExecution structure. I agree that it doesn't belong in the Poll request but this was the conclusion from the meeting we had yesterday. I have shared the doc with you, I can send it again as well. Do you mind taking a look and adding your comments there?